staging/src/k8s.io/kubectl/pkg/cmd/run/run_test.go

```diff
@@ -398,7 +398,7 @@ func TestGenerateService(t *testing.T) {
 				test.params["port"] = test.port
 			}
 
-			_, err = opts.generateService(tf, cmd, test.serviceGenerator, test.params, "test")
+			_, err = opts.generateService(tf, cmd, test.serviceGenerator, test.params, "test", true)
```

Comment:
I feel like this change loses the old test case.  Can the test be performed for both true and false, to make sure both paths are covered?